### PR TITLE
`torchtyping` => `jaxtyping`

### PR DESCRIPTION
**Pull request by brentyi**
Draft at migrating from `torchtyping` to `jaxtyping`. This will fix static type checking.

Some notes:
- This will require bumping the minimum Python version to 3.8.
- I'm currently using `jaxtyping.Shaped[]` everywhere, but we should ideally be using `jaxtyping.Float[]`, `jaxtyping.Int[]`, etc.

cc @tancik

### CODE REVIEW

File: _torchtyping_to_jaxtyping.py
Comment (tancik) [inline _torchtyping_to_jaxtyping.py:1]: Should we keep this around in the scripts folder if others want to update their code?

Comment (brentyi) [reply _torchtyping_to_jaxtyping.py:1]: Nah let's not, I think it doesn't add much because changes still require so much manual labor

**Review by tancik — APPROVED** [review]
LGTM

**Comment by brentyi** [comment]
Conversion script we used, for posterity: https://gist.github.com/brentyi/21d08fe4e90e0397b9484893cf1b9822

**Comment by brentyi** [comment]
🎊 🎊 🎊 🎊 🎊 

cc @patrick-kidger thanks for adding torch support to your library :)

**Comment by patrick-kidger** [comment]
Thank you for the tag! This is really great to see. :)

**Pull request merged**